### DISCUSSION TITLE
Turborepo for parallel merge queues

### DISCUSSION THREAD

**Discussion by debkanchan** [trunk-io/docs#310] [Feature Requests]
We have Nx so why not. IDK i don't really know about turborepo in depth tho

**Comment by pv72895** [comment]
Hey DK! Thanks for making this discussion - it helps us gauge interest in supporting more integration like this

Just to scope this and for others reading this, I'm taking this feature request to mean providing first-class support for using parallel queues with Merge + Turborepo by providing a GH action like we do for bazel and Nx. Merge does currently [support any CI system](https://docs.trunk.io/merge/parallel-queues/api) with respect to [parallel merge queues](https://docs.trunk.io/merge/parallel-queues) - we expose a public API that can be used to tell merge what specifically the PR you're submitting to the queue impacts

Right now though, Turborepo users have to figure that out for themselves, most likely through a combination of [--graph](https://turbo.build/repo/docs/reference/run#--graph-file-type) to get the dependency graph and [--filter](https://turbo.build/repo/docs/crafting-your-repository/running-tasks#filtering-by-source-control-changes) to compare changes between a raised PR and the current view of `main`. Merge does have plans to eventually add first-class support for this with a Turborepo GH action, but that's not for the near-term future. Is there anything specific you'd like to see with Turborepo support in Merge or is this just to round merge out?

**Reply by debkanchan** [reply]
just to round merge out
